spec/unit/recipes/default_spec.rb

```diff
@@ -6,9 +6,46 @@
       cached(:chef_run) do
         ChefSpec::SoloRunner.new(p).converge(described_recipe)
       end
+
+      include_context 'common_stubs'
+
       it 'converges successfully' do
         expect { chef_run }.to_not raise_error
       end
+      [
+        '# Configured as client',
+        'myorigin = $mydomain',
+        'relayhost = [smtp.osuosl.org]:25',
+        'smtpd_use_tls = no',
+        'smtp_use_tls = no',
+      ].each do |line|
+        it do
+          expect(chef_run).to render_file('/etc/postfix/main.cf').with_content(line)
+        end
+      end
+      context 'postfix submission port' do
+        cached(:chef_run) do
+          ChefSpec::SoloRunner.new(p) do |node|
+            node.automatic['network']['default_gateway'] = '10.162.136.1'
+          end.converge(described_recipe)
+        end
+        [
+          'relayhost = [smtp.osuosl.org]:587',
+          'smtp_use_tls = yes',
+        ].each do |line|
+          it do
+            expect(chef_run).to render_file('/etc/postfix/main.cf').with_content(line)
+          end
+        end
+        if p == DEBIAN_8
+          it do
+            expect(chef_run).to render_file('/etc/postfix/main.cf').with_content('smtp_tls_CAfile = /etc/ssl/certs/ca-certificates.crt')
```

Comment:
Add an else here and check that for all other platforms, this should *NOT* happen.